app/models/post.rb

```diff
@@ -1,3 +1,108 @@
+require 'rest-client'
+
 class Post < ApplicationRecord
-  belongs_to :user
+  belongs_to :user  
+  # firebase_url    = 'entheogen-a76f2.firebaseapp.com'
+
+  def calculate_running_averages(array)
+    arr = array.map.with_index do |val, id|
+      if id > 0 
+          pre_avg = array[id-1]
+          pre_sum = pre_avg * id
+          len = id+1
+          (val+pre_sum)/len
+      else 
+          val
+      end 
+    end
+  end
+
+  #function to scale events between two numbers
+  #reference - https://cycling74.com/forums/what's-the-math-behind-the-scale-object
+
+  def scale(x, xmin, xmax, ymin, ymax)
+    xrange = xmax - xmin
+    yrange = ymax - ymin
+    ymin + (x - xmin) * (yrange.to_f / xrange)     
+  end
+
+  #get trough and peak for each average value
+  # p stats
+  def get_relative_min_max(array)
+      min = 1.00000000000000000
+      max = 0.00000000000000000
+      arr = []
+      test = array.map.with_index do |val, id|
+          # relative_minimum = min
+          # relative_maximum = max
+
+          if  val < min 
+              min = val    
+          elsif val > max
+              max = val
+          elsif val == min && val == max 
+            min = val - 0.00000000000000001
+            max = val
+          end
+    
+          r = scale(val, min, max, -160, -20)
+          d = scale(val, min, max, 300, 500)
+          n = scale(val, min, max, 20, 100)
+
+          
+          {
+              average: val,
+              rel_min: min,
+              rel_max: max,
+              scaled_dist: d,
+              scaled_rad: r,
+              scaled_noise: n 
+              # scaled_dist: scale(val, min, max, 20, 600),
+              # scaled_rad: scale(val, min, max, -160, 100) 
+          }    
+      end
+  end
+
+  def eeg
+    # byebug
+     response = RestClient.get self.data_file_url
+     orig_data = response.body
+    # byebug
+     new_data = orig_data.gsub!("NaN", "0.0")
```

Comment:
remove "NaN"s from data
example of data: 
https://firebasestorage.googleapis.com/v0/b/entheogen-a76f2.appspot.com/o/eeg%2F2268f66c-336f-4521-ba48-aa196a2787c0.json?alt=media&token=b209fa87-16a5-44b0-9ea1-4cc16f745aec